lib/r10k/environment/base.rb

```diff
@@ -124,28 +151,37 @@ def accept(visitor)
   end
 
   def deploy
-    puppetfile.load(@overrides.dig(:environments, :default_branch_override))
+    loaded_content = @loader.load
+    @base_modules = loaded_content[:modules]
 
-    puppetfile.sync
```

Comment:
It might make this PR more understandable to do the content sync in one commit (by getting the modules from the puppetfile object. Then adding loader in another commit.